packages/core/src/render3/definition.ts

```diff
@@ -755,3 +753,36 @@ export function getNgModuleDef<T>(type: any, throwNotFound?: boolean): NgModuleD
   }
   return ngModuleDef;
 }
+
+/**
+ * A method can returns a component ID from the component selectors.
+ */
+function getComponentId(componentDef: Partial<ComponentDef<unknown>>): string {
+  let hash = 0;
+
+  const hashSelectors = [
+    ...componentDef.selectors || [],
```

Comment:
nit: I think it'd help with readability:
```suggestion
    ...(componentDef.selectors || []),
```